File: BaekJoon/Number Theory & Combinatorics/2609.py
import sys


x, y = map(int, sys.stdin.readline().split())
a, b = x, y

gcd = None
lcm = None

while True :
    if a == 1 or b == 1:
        gcd = 1
        break
    if a > b:
        if a%b ==0:
            gcd = b
            break
        else:
            a %= b
    else:
        if b%a ==0:
            gcd = a
            break
        else :
            b %= a

lcm = x * y // gcd

print(gcd)
print(lcm)

File: BaekJoonSwift/BaekJoonSwift/Numbers and Combinations/2609.swift
import Foundation

var i = readLine()!.split(separator: " ").map { Int($0)! }
var lcm = i[0]*i[1]
while i[0] != 0 && i[1] != 0 {
    if i[0] > i[1] {
        i[0] %= i[1]
    } else {
        i[1] %= i[0]
    }
}

let gcd = i.max()!
lcm /= gcd
print(i.max()!)
print(lcm)

